src/components/Trade/SelectAsset.tsx

```diff
@@ -8,17 +7,12 @@ import { Text } from 'components/Text'
 
 import { WithBackButton } from './WithBackButton'
 
-type SelectAssetProps = { onClick: (asset: Asset) => void } & RouteComponentProps
-
-export const SelectAsset = ({ onClick, history }: SelectAssetProps) => {
-  // Filters the asset search to only show eth/erc20 assets
-  const ethCAIP2 = caip2.toCAIP2({ chain: ChainTypes.Ethereum, network: NetworkTypes.MAINNET })
-  const filterByCaip19 = (assets: Asset[]): Asset[] => {
-    return assets.filter(
-      ({ chain, network }: Asset) => caip2.toCAIP2({ chain, network }) === ethCAIP2,
-    )
-  }
+type SelectAssetProps = {
+  onClick: (asset: Asset) => void
+  filterBy: (assets: Asset[]) => Asset[]
```

Comment:
as we talked about, this is a smell, because of SelectAsset, AssetSearch, and AssetList downstream components.

ids should be passed down as far as possible, and at the last minute the view layer should get the actual Asset object from the id

let's make a ticket and address this later. 

additionally, each swapper should compute and cache supported assetIds once and just return the cache so it's O(n) once then O(1) after

Reply:
please make another ticket to simplify this stuff further

Reply:
after we digged further we realized we can't assume the assetIds do not change - as we're about to start adding assets to the store dynamically - so please consider that in the context of my comment above, it may no longer be valid to initialize and cache and maybe this O(n) is fine